api/common/services/RoleService.ts

```diff
@@ -14,7 +22,50 @@ export class RoleService {
     if (existing.length > 0) {
       const asRoleType = Object.assign(new Role(), existing[0]);
       return { exists: true, role: asRoleType };
-    } 
-    return { exists: false, role: null };
+    }
+    return { exists: false };
+  }
+
+  public async createRole(request: RoleCreationRequest): Promise<Role> {
+    const role = Role.fromRequest(request);
+    const { success, key } = await this.createKey(request.name);
+    if (!success) {
+      return undefined;
+    }
+    role.apiKey = key;
+    await this._repo.saveOrUpdate<Role>(role);
+    return role;
+  }
+
+  private async createKey(name: string): Promise<{ success: boolean, key?: string }> {
+    let url = `https://control.ably.net/v1/apps/${process.env.APP_ID}/keys`;
+    
+    let body = JSON.stringify({
+      name: name,
+      // TODO: Replace with inserted capabilities
+      capability: {
+        '*': ["publish", "subscribe", "presence"]
+      }
+    });
+
+    const headers = { 
+      "Content-type": "application/json",
+      "accept": "application/json",
+      "Authorization": "Bearer " + process.env.CONTROL_KEY,
+      "Access-Control-Allow-Origin": "*"
+    };
+
+    const response = await fetch(url, {method: 'POST', body, headers });
+    if (response.status != 201) {
+      return { success: false };
+    }
+    const responseJson = await response.json();
+    const key = responseJson["key"];
+
+    if (!key) {
```

Comment:
This could probably be a single return { success, key } even where the key is undefined.